### PR TITLE
Feature/33 comment query

### PR DESCRIPTION
**Pull request by oddevan**
In support of #33 and #60

### Link

[Test branch on Vercel](https://nextjs-wordpress-starter-git-feature-33-comment-query.webdevstudios.vercel.app/)

### Description

Add paginated query for getting comments per article. Also add comment fields to post and page queries.

### Screenshot

Screen Shot 2021-01-21 at 4 50 52 PM

### Verification

1. Visit a page. See the empty JSON shown in the picture.
1. For additional verification: import the [WordPress theme unit test](https://codex.wordpress.org/Theme_Unit_Test)
1. Visit the "Template: Comments" post at http://localhost:3000/blog/template-comments
2. Note the visible JSON for the comments on the post

### CODE REVIEW

**Review by ravewebdev — COMMENTED** [review]
Looks good to me! My only question would be whether we intend to include comments on pages at all, or just posts for now?

**Review by gregrickaby — CHANGES_REQUESTED** [review]
@oddevan Nicely done. I agree with @ravewebdev on this... can you yank the comment display/query out of pages? I think posts is fine for now.

Thank you!

File: components/blocks/BlockHeadings/BlockHeadings.js [hunk 1]
```diff
@@ -9,7 +9,8 @@ import React from 'react'
  * The core Headings block from Gutenberg.
  *
  * @author WebDevStudios
- * @param {object} props The component attributes as props.
+ * @param props.props
+ * @param {object}    props The component attributes as props.
```
Comment (ravewebdev) [inline components/blocks/BlockHeadings/BlockHeadings.js:13]: This should probably be something like:
```
@param {object} props           The component attributes as props.
@param {object} props.props The block attributes props from WordPress.
```

File: components/blocks/BlockParagraph/BlockParagraph.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ import PropTypes from 'prop-types'
  * The core Paragraph block from Gutenberg.
  *
  * @author WebDevStudios
- * @param {object} props The component attributes as props.
+ * @param props.props
+ * @param {object}    props The component attributes as props.
```
Comment (ravewebdev) [inline components/blocks/BlockParagraph/BlockParagraph.js:12]: Same comment issue as above

**Review by ravewebdev — CHANGES_REQUESTED** [review]
Looks good - just seeing some weird JSDocs on components.

File: components/blocks/BlockHeadings/BlockHeadings.js [hunk 3]
```diff
@@ -9,7 +9,9 @@ import React from 'react'
  * The core Headings block from Gutenberg.
  *
  * @author WebDevStudios
- * @param {object} props The component attributes as props.
+ * @param {object} props       The component attributes as props.
+ * @param {object} props.props The block attributes props from WordPress.
+ * @return {Element}           RichText component with heading content
```
Comment (ravewebdev) [inline components/blocks/BlockHeadings/BlockHeadings.js:14]: One last thing - missing a period at the end of the `@return` comment.

File: components/blocks/BlockParagraph/BlockParagraph.js [hunk 4]
```diff
@@ -8,7 +8,9 @@ import PropTypes from 'prop-types'
  * The core Paragraph block from Gutenberg.
  *
  * @author WebDevStudios
- * @param {object} props The component attributes as props.
+ * @param {object} props       The component attributes as props.
+ * @param {object} props.props The block attributes props from WordPress.
+ * @return {Element}           RichText component with paragraph content
```
Comment (ravewebdev) [inline components/blocks/BlockParagraph/BlockParagraph.js:13]: Same as above.

**Review by ravewebdev — CHANGES_REQUESTED** [review]

**Review by ravewebdev — APPROVED** [review]

**Review by gregrickaby — APPROVED** [review]

**Pull request merged**